src/common/KokkosKernels_IOUtils.hpp

```diff
@@ -180,26 +180,45 @@ void kk_diagonally_dominant_sparseMatrix_generate(
   {
     int varianz = (1.0*rand()/RAND_MAX-0.5)*row_size_variance;
     rowPtr[row+1] = rowPtr[row] + elements_per_row+varianz;
+    if(rowPtr[row+1] == rowPtr[row])   // This makes sure that
+      rowPtr[row+1]++;                 // there are no empty rows
```

Comment:
@brian-kelley these two lines make sure that `rowPtr[row+1] >= rowPtr[row]+1` even though `elements_per_row+varianz` happens to be zero. This was the easiest solution I came up with.

Reply:
@seheracer But if row_size_variance is very big, and rand() returns close to 0, I'm worried that ``(1.0*rand()/RAND_MAX-0.5)`` will be about -0.5, so ``elements_per_row + varianz`` could be negative. Then ``rowPtr[row+1] < rowPtr[row]``.

Reply:
I just remembered that I did this fix a while ago for another function kk_sparseMatrix_generate() at the top of IOUtils. Should have looked for other places where it was needed.

Reply:
I see what you mean now. Good catch! I should change lines 183-184 to
```
if(rowPtr[row+1] <= rowPtr[row]) 
  rowPtr[row+1] = rowPtr[row] + 1; 
```

Reply:
Yes that will fix it. Thanks, I'll approve when that's committed.